Show 'Display visitor groups in dashboard' toggle even if GA is disco…

ankitrox opened the pull request:
…nnected.

## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9264 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

benbowler reviewed (approved):
Looks good @ankitrox.

E2E test are failing due to the regression outlined to be fixed in #9225.

hussain-t reviewed (approved):
Great work, @ankitrox! LGTM 👍

Just a heads-up, I made a minor adjustment to rearrange the import and updated the QAB to ensure that Audience Segmentation is enabled to configure the audiences. Without this, the configured audiences would return as `null`.

I've verified that the test failures are unrelated to this PR.

ankitrox commented:
Thank you @hussain-t 

I've addressed the issues in E2E and there are no relevant errors.

I can see that JS tests, VRT are failing in pipeline, but getting passed in local, so moving this to CR.

hussain-t reviewed (approved):
Nice work, @ankitrox. LGTM 👍

techanvil commented on assets/js/components/settings/SettingsAdmin.js:
```diff
@@ -43,6 +44,9 @@ import { useFeature } from '../../hooks/useFeature';
 export default function SettingsAdmin() {
 	const audienceSegmentationEnabled = useFeature( 'audienceSegmentation' );
 
+	const configuredAudiences = useSelect( ( select ) =>
+		select( CORE_USER ).getConfiguredAudiences()
```
Hi @ankitrox, this change has introduced a regression as reported by @wpdarren in [Slack](https://fueled10up.slack.com/archives/CBKKQEBR9/p1726753396158329).

We need to guard against the request to the `audience-settings` endpoint with a check for `audienceSegmentationEnabled` here:

```suggestion
	const configuredAudiences = useSelect(
		( select ) =>
			audienceSegmentationEnabled &&
			select( CORE_USER ).getConfiguredAudiences()
```

Please can you raise a followup PR to fix it, targeting the `main` branch?
